apps/web/src/views/BuyCrypto/hooks/usePriceQuoter.ts

```diff
@@ -1,164 +1,67 @@
-  provider: string
-}
-
-type ProviderResponse = (Partial<MercuryoQuote> & Partial<PriceQuotes> & Partial<BinanceConnectQuote>) | undefined
-
-export interface ProviderAvailabilityData {
-  MoonPay: boolean
-  Mercuryo: boolean
-  BinanceConnect: boolean
-}
-
-const calculateQuotesData = (quote: PriceQuotes): ProviderQoute => {
-  return {
-    providerFee: quote.feeAmount,
-    networkFee: quote.networkFeeAmount,
-    amount: quote.quoteCurrencyAmount,
-    quote: quote.quoteCurrencyPrice,
-    provider: 'MoonPay',
-  }
-}
-
-const calculateQuotesDataMercury = (quote: MercuryoQuote, inputCurrency: string): ProviderQoute => {
-  return {
-    providerFee: Number(quote.data.fee[inputCurrency.toUpperCase()]),
-    networkFee: 0,
-    amount: Number(quote.data.amount),
-    quote: Number(quote.data.rate),
-    provider: 'Mercuryo',
-  }
-}
-
-const calculateQuotesDataBsc = (quote: BscQuote): ProviderQoute => {
-  return {
-    providerFee: quote.userFee,
-    networkFee: quote.networkFee,
-    amount: quote.cryptoAmount,
-    quote: quote.quotePrice,
-    provider: 'BinanceConnect',
-  }
-}
-
-const calculateNoQuoteOption = (quote: ProviderResponse): ProviderQoute => {
-  let provider = 'MoonPay'
-  if (quote?.code === 'OPE100000031') provider = 'BinanceConnect'
-  else if (quote?.code === 'ERR_BAD_REQUEST') provider = 'Mercuryo'
-  return {
-    providerFee: 0,
-    networkFee: 0,
-    amount: 0,
-    quote: 0,
-    provider,
-  }
-}
-
-const usePriceQuotes = (amount: string, inputCurrency: string, outputCurrency: string, userIp: string | null) => {
+import { Field } from 'state/buyCrypto/actions'
+import { useBuyCryptoState } from 'state/buyCrypto/hooks'
+import { fetchProviderQuotes } from './useProviderQuotes'
+import { fetchProviderAvailabilities } from './useProviderAvailability'
+import { chainIdToNetwork } from '../constants'
+import { ProviderQoute } from '../types'
+
+const usePriceQuotes = () => {
   const [quotes, setQuotes] = useState<ProviderQoute[]>([])
   const { chainId } = useActiveChainId()
 
-  const fetchProviderAvailability = useCallback(
-    async (ip: string, combinedData: ProviderQoute[]) => {
-      // first check user availability
-      const responsePromises = [fetchMoonpayAvailability(ip), fetchMercuryoAvailability(ip)]
-      const responses = await Promise.allSettled(responsePromises)
-
-      const dataPromises = responses.reduce((accumulator, response) => {
-        if (response.status === 'fulfilled') {
-          return [...accumulator, response.value]
+  const {
+    typedValue: amount,
+    [Field.INPUT]: { currencyId: inputCurrency },
+    [Field.OUTPUT]: { currencyId: outputCurrency },
+    userIpAddress: userIp,
+  } = useBuyCryptoState()
+
+  const sortProviderQuotes = useCallback(
+    async (combinedData: ProviderQoute[]) => {
+      let sortedFilteredQuotes = combinedData
+      try {
+        if (userIp) {
+          const providerAvailabilities = await fetchProviderAvailabilities({ userIp })
```

Comment:
Does the availability of the provider change constantly? If not we could have a separate state like `availableProviders`? No need to fetch every time

Reply:
i could move this to the buyCrypto redux state however, the reason i have it here is the availability can change depending on what crypto the user is trying to purchase. Saying that i think for the current assets we support i dont think there is any outliers. so will i remove then?.

Reply:
If the availability depends on the asset, why it's not part of the input parameter? The function signature to me now looks like only depend on the user ip.

Also there's no need to move to redux store. We can use `react-query`. You can specify the dependencies as the query key and it won't refetch unless the query key changes.

Imo, if we can use hook or component level state to solve the problem, there's no need to use redux. But it's just my preference. I'm open to further discussion regarding this topic

Reply:
resolved, including in pr for add wbtc as a new currency